release/ray_release/result.py

```diff
@@ -81,7 +81,24 @@ class ExitCode(enum.Enum):
     PREPARE_ERROR = 43
 
 
-def handle_exception(e: Exception) -> Tuple[ExitCode, BuildkiteExitCode, Optional[int]]:
+def _is_transient_error(runtime: int) -> bool:
+    """
+    Classify whether an infra-failure issue is a transient issue. This is based on
+    the status of its previous retries, and its runtime.
+    """
+    retry_count = int(os.environ.get("BUILDKITE_RETRY_COUNT", "0"))
+    if retry_count > 0:
+        # Already retried at least once and failed again, not a transient issue
+        return False
```

Comment:
so this is like, if it retries and failed for more than once, then it is not a "transicent error"?

how many "transcient error" do we have today and what are they?

like maybe we can just say "all transcient errors are now infr errors" today? :)

Reply:
transient error is a concept I introduced in a previous PR: https://github.com/ray-project/ray/pull/34057. The idea is that, anything that is classified as infra error by test runner is classified as a transient error first, so buildkite will retry it. But if the retry failed, classify them as infra error so buildkite won't retry them again.